contracts/src/v0.8/dev/interfaces/OCR2DRBillableInterface.sol

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.6;
+
+import "./OCR2DRRegistryInterface.sol";
+
+/**
+ * @title OCR2DR billable oracle interface.
+ */
+interface OCR2DRBillableInterface {
+  /**
+   * @notice Determine the fee charged by the DON that will be paid to Node Operators for servicing a request
+   * @param data Encoded OCR2DR request data, use OCR2DRClient API to encode a request
+   * @param billing The request's billing configuration
+   * @return fee Cost in Juels (1e18) of LINK
+   */
+  function getRequiredFee(bytes calldata data, OCR2DRRegistryInterface.RequestBilling calldata billing)
+    external
+    returns (uint96);
+
+  /**
+   * @notice Estimate the total cost that will be charged to a subscription to make a request: gas re-imbursement, plus DON fee, plus Registry fee
+   * @param data Encoded OCR2DR request data, use OCR2DRClient API to encode a request
+   * @param billing The request's billing configuration
+   * @return billedCost Cost in Juels (1e18) of LINK
+   */
+  function estimateCost(bytes calldata data, OCR2DRRegistryInterface.RequestBilling calldata billing)
```

Comment:
How do we estimate cost given the returned value or error may contain a varying # of bytes?

Reply:
My understanding is, we can only estimate cost by providing a "dummy" return value in order to accurately estimate the amount of gas that will be used to call the callback.

Could we provide an "estimateCost" view function that allows a user to supply their expected return value to more accurately calculate the estimated cost?

Reply:
Else, I suppose we could just assume that the full `gasLimit` amount of gas is used for estimating cost.  Thoughts?

Reply:
Currently there is a very naive approach of keeping
```
    uint32 gasOverhead;
```
in the Registry config. This is meant to represent the "average" cost. This along with the max `gasLimit` provides an estimate, not exact guarantee.

I do agree that we could make this smarter. I'll think about it more. Might require playing around with it to dial in.

Reply:
TODO to myself:
Add tests for gas cost given large returned value/error.